colossus/src/main/scala/colossus/service/ServiceClient.scala

```diff
@@ -160,7 +160,6 @@ class ServiceClient[P <: Protocol](
     val context: Context
 )(implicit tagDecorator: TagDecorator[P] = TagDecorator.default[P])
     extends ControllerDownstream[Encoding.Client[P]]
-    with HasUpstream[ControllerUpstream[Encoding.Client[P]]]
```

Comment:
This seems ok since it's very small, but in the future I would like to avoid making changes that aren't relevant to the issue being addressed in the PR.